examples/acados_python/pendulum_on_cart/ocp/example_hessian_regularization.py

```diff
@@ -5,6 +5,28 @@
 from acados_template import AcadosOcp, AcadosOcpSolver
 from pendulum_model import export_pendulum_ode_model
 
+def get_iterate_hessian(ocp_solver: AcadosOcpSolver, N) -> list:
+    """
+    Obtain exact Hessian from the current iterate of the ocp_solver of every stage
+    and form the block diagonal Hessian matrix.
+    """
+    exact_Hessian_list = []
+    for i in range(N+1):
+        hess_block = ocp_solver.get_hessian_block(i)
+        exact_Hessian_list.append(hess_block)
+    return exact_Hessian_list
+
+def check_positive_definite(whole_Hessian, N):
```

Comment:
Ist this needed? Could ne added to qp diagnostics?

Reply:
Better just use qp diagnostic and check the eigenvalues obtained there IMO